src/UI/Core/Label.cpp

```diff
@@ -0,0 +1,41 @@
+#include "Label.h"
+
+#include "../../Constants.h"
+#include "../../FontManager.h"
+
+#include "NAS2D/Renderer/Renderer.h"
+#include "NAS2D/Resources/Font.h"
+
+#include "NAS2D/MathUtils.h"
+#include "NAS2D/Utility.h"
+
+#include <locale>
+
+static const int FIELD_PADDING = 4;
+static NAS2D::Font* TXT_FONT = nullptr;
+
+Label::Label() {
+    TXT_FONT = NAS2D::Utility<FontManager>::get().font(constants::FONT_PRIMARY, constants::FONT_PRIMARY_NORMAL);
+    height(static_cast<float>(TXT_FONT->height() + FIELD_PADDING * 2));
+}
+
+Label::~Label() {
+
+}
```

Comment:
Can this be set as `default` in the header for a `virtual` method? I suspect that would be possible.

Reply:
It should be as long as it's declared virtual (though I suspect that only really matters in the parent class)

Reply:
Oh yeah, I suppose the destructor could just be removed if it's not actually being overridden. Ehh, minor point.